### PR TITLE
GN-4748: basic support for fetching dynamic decision templates

### PR DESCRIPTION
**Pull request by elpoelma**
### Overview
This PR adds basic support for fetching dynamic decision templates from the regulatory statement endpoint.
Specifically it includes the following changes/additions:

**A refactor of the `RegulatoryStatementsFetcher` service**
- Renamed to `TemplateFetcher` to better reflect its purpose
- Add support for fetching templates of different types (not only regulatory statement templates)
- Add support for querying the `contexts`/`disabledInContexts` properties of templates (important for decisions/agendapoints)

**Modifications to models of agendapoint routes**
- Modified `new` route of agendapoints to also fetch decision templates from regulatory statement endpoint
- Modified `edit` route of agendapoints to also fetch decision templates from regulatory statement endpoint

I made the following initial assumptions in this PR:
- The type of a decision template is `http://data.lblod.info/vocabularies/gelinktnotuleren/BesluitTemplate`
- The predicates used to specify template contexts are: `ext:context` and `ext:disabledInContext`

@abeforgit section:

** Modifications to support new structure node, and the removal of the decision nodes in version 22 of the plugins
** bumps and config changes to support v10 editor and v22 plugins
- merge of @piemonkey 's PR to get the deps to be friends
- changed the builder image cause our ember images don't have a node v20 version

##### connected issues and PRs:
[GN-4906](https://binnenland.atlassian.net/browse/GN-4906)
[GN-4748](https://binnenland.atlassian.net/browse/GN-4748?atlOrigin=eyJpIjoiMWM0ZjhiYzU0ZjZmNDYyYmIwNGIzZDM4NDM5YzdlNWIiLCJwIjoiaiJ9)
https://github.com/lblod/frontend-reglementaire-bijlage/pull/236
https://github.com/lblod/app-reglementaire-bijlage/pull/77
https://github.com/lblod/reglement-publish-service/pull/14
Solves https://github.com/lblod/frontend-reglementaire-bijlage/issues/227

### Setup
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/77 for a description on how to set-up the stack

### How to test/reproduce
Check-out https://github.com/lblod/app-reglementaire-bijlage/pull/77 for a description on how to test the new features.

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check cancel/go-back flows
- [x] Check database state correct when deleting/updating (especially regarding relationships)
- [x] changelog
- [x] npm lint
- [x] no new deprecations

### CODE REVIEW

File: app/routes/agendapoints/edit.js [hunk 1]
```diff
@@ -18,12 +19,17 @@ export default class AgendapointsEditRoute extends Route {
   async model() {
     const { documentContainer, returnToMeeting } =
       this.modelFor('agendapoints');
-    const standardTemplates = this.standardTemplate.fetchTemplates.perform();
+    const standardTemplates =
+      await this.standardTemplate.fetchTemplates.perform();
```
Comment (piemonkey) [inline app/routes/agendapoints/edit.js:23]: `await`s here will mean the requests run in series, I think we can remove the awaits and just pass the promises to `RSVP` so they run in parallel.

File: app/routes/inbox/agendapoints/new.js [hunk 2]
```diff
@@ -14,8 +15,14 @@ export default class InboxAgendapointsNewRoute extends Route {
...
-    ]);
+    const standardTemplates = this.standardTemplate.templatesForContext(
+      templates,
+      ['http://data.vlaanderen.be/ns/besluit#BehandelingVanAgendapunt'],
+    );
+    const dynamicTemplates = await this.templateFetcher.fetch.perform({
+      templateType:
+        'http://data.lblod.info/vocabularies/gelinktnotuleren/BesluitTemplate',
+    });
+    return [...standardTemplates, ...dynamicTemplates];
```
Comment (piemonkey) [inline app/routes/inbox/agendapoints/new.js:26]: It would be nice to do this as one `await` with `Promise.all()`.

**Review by piemonkey — APPROVED** [review]
Only minor comments that could speed things up, otherwise good.

File: app/routes/agendapoints/edit.js (hunk 1 above)
Comment (elpoelma) [reply app/routes/agendapoints/edit.js]: Done :)

**Comment by abeforgit** [comment]
ok now the commits are on the right branch lol

**Pull request merged**